string-manipulation-3/src/is-anagram.js

```diff
@@ -0,0 +1,29 @@
+/* exported isAnagram */
+function isAnagram(firstString, secondString) {
+  var firstStringArray = firstString.split('');
+  var secondStringArray = secondString.split('');
+  var hasASpace = true;
+  firstStringArray.sort();
+  secondStringArray.sort();
+  while (hasASpace) {
+    if (firstStringArray[0] === ' ') {
+      firstStringArray.shift();
+    } else {
+      hasASpace = false;
+    }
+  }
+  hasASpace = true;
+  while (hasASpace) {
+    if (secondStringArray[0] === ' ') {
+      secondStringArray.shift();
+    } else {
+      hasASpace = false;
+    }
+  }
+  for (var i = 0; i < firstString.length; i++) {
+    if (secondStringArray[i] !== firstStringArray[i]) {
+      return false;
+    }
+  }
```

Comment:
You can simplify this further by turning the arrays back into strings once they are sorted. There is then a built in string method you can use to `trim` any whitespace from the beginning of the string. Once the strings are sorted and the whitespace removed you can just check if they are equal to each other or not. That would eliminate the need for the 2 `while` loops and the final `for` loop.